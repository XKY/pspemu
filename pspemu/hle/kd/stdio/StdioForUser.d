module pspemu.hle.kd.stdio.StdioForUser; // kd/stdio.prx (sceStdio)

import pspemu.hle.ModuleNative;

class StdioForUser : ModuleNative {
	void initNids() {
		mixin(registerd!(0x172D316E, sceKernelStdin));
		mixin(registerd!(0xA6BAB2E9, sceKernelStdout));
		mixin(registerd!(0xF78BA90A, sceKernelStderr));
		mixin(registerd!(0x98220F3E, sceKernelStdoutReopen));
		mixin(registerd!(0xFB5380C5, sceKernelStderrReopen));
	}

	/**
	 * Function to get the current standard in file no
	 * 
	 * @return The stdin fileno
	 */
	SceUID sceKernelStdin() { return STDIN; }

	/**
	 * Function to get the current standard out file no
	 * 
	 * @return The stdout fileno
	 */
	SceUID sceKernelStdout() { return STDOUT; }

	/**
	 * Function to get the current standard err file no
	 * 
	 * @return The stderr fileno
	 */
	SceUID sceKernelStderr() { return STDERR; }

	/** 
	 * Function reopen the stdout file handle to a new file
	 *
	 * @param file - The file to open.
	 * @param flags - The open flags 
	 * @param mode - The file mode
	 * 
	 * @return < 0 on error.
	 */
	int sceKernelStdoutReopen(string file, int flags, SceMode mode) {
		unimplemented();
		return -1;
	}

	/** 
	 * Function reopen the stderr file handle to a new file
	 *
	 * @param file - The file to open.
	 * @param flags - The open flags 
	 * @param mode - The file mode
	 * 
	 * @return < 0 on error.
	 */
	int sceKernelStderrReopen(string file, int flags, SceMode mode) {
		unimplemented();
		return -1;
	}
}


enum : SceUID {
	STDIN  = -1,
	STDOUT = -2,
	STDERR = -3
}

static this() {
	mixin(ModuleNative.registerModule("StdioForUser"));
}