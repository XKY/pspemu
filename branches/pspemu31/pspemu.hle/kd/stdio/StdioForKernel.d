module pspemu.hle.kd.stdio.StdioForKernel; // kd/stdio.prx (sceStdio)

import pspemu.hle.ModuleNative;
import pspemu.hle.kd.stdio.StdioForUser;

class StdioForKernel : StdioForUser {
}

static this() {
	mixin(ModuleNative.registerModule("StdioForKernel"));
}