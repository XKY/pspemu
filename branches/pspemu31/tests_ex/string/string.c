//#pragma compile, "%PSPSDK%/bin/psp-gcc" -I. -I"%PSPSDK%/psp/sdk/include" -L. -L"%PSPSDK%/psp/sdk/lib" -D_PSP_FW_VERSION=150 -Wall -g -O0 string.c ../common/emits.c -lpspsdk -lc -lpspuser -lpspkernel -o test.elf
//#pragma compile, "%PSPSDK%/bin/psp-fixup-imports" string.elf

#include <pspkernel.h>
#include <pspthreadman.h>
#include <pspdebug.h>
#include <stdio.h>
#include <string.h>
#include <../emits.h>

PSP_MODULE_INFO("THREAD TEST", 0, 1, 1);
PSP_MAIN_THREAD_ATTR(THREAD_ATTR_USER | THREAD_ATTR_VFPU);

int main(int argc, char** argv) {
	char buffer[128];
	
	pspDebugScreenInit();
	
	sprintf(buffer, "%d", (u32)0);
	emitString(buffer);
	pspDebugScreenPrintf("%s\n", buffer);
	
	sprintf(buffer, "%d", (u32)100000);
	emitString(buffer);
	pspDebugScreenPrintf("%s\n", buffer);
	
	sprintf(buffer, "%lld", (u64)100000);
	emitString(buffer);
	pspDebugScreenPrintf("%s\n", buffer);

	sceKernelExitGame();
	return 0;
}
