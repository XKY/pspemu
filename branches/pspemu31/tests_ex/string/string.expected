0
100000
9
100000